### PR TITLE
Use library's own testing API

### PR DESCRIPTION
**Pull request by colriot**
* Use own testing API explicitly instead of basic `kotlin-test` asserts
* Move `assertSuccess` from implicit invocation in `value` getter to explicit usage in actual test methods. Making logic/intention explicit

### CODE REVIEW

File: reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/maybe/TestMaybeObserver.kt [hunk 1]
```diff
@@ -9,13 +9,7 @@ import com.badoo.reaktive.utils.freeze
 class TestMaybeObserver<T>(autoFreeze: Boolean = true) : TestObserver(), MaybeObserver<T> {
 
     private val _value = AtomicReference<Value<T>?>(null)
-
-    val value: T
-        get() {
-            assertSuccess()
```
Comment (arkivanov) [inline reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/maybe/TestMaybeObserver.kt]: Why did you remove the assert? Now if `Maybe` not succeeded then NPE will be thrown which is much less understandable. This is the public API.

Comment (arkivanov) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/maybe/TestMaybeObserver.kt]: I think it's better to keep it, in case if a client will just read the value without `assertSuccess`.

Comment (colriot) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/maybe/TestMaybeObserver.kt]: Because that brings an implicit assumption that if you access `value` then some `assert` will be done for you (API user). And with tests it's always better to be explicit so that the test method itself will contain all the meaningful logic in its body. I.e. if a user assumes successful completion, they should assert it with `assertSuccess` explicitly.

I agree with non-clear NPE argument though. I'd use `requireNonNull(..., "Should 'assertSuccess' before accessing a 'value'")` which will explain and encourage explicit use of API. 
Wdyt?

Comment (arkivanov) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/maybe/TestMaybeObserver.kt]: I'm fine with requireNonNull, but I would like message to be more clear, like "Single did not success" or something. So it will be clear why there is no value. @CherryPerry what do you think?

Comment (colriot) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/maybe/TestMaybeObserver.kt]: How about "Single did not succeed. Assert that with 'assertSuccess()' before accessing a 'value'"?

Comment (arkivanov) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/maybe/TestMaybeObserver.kt]: Works for me. @CherryPerry ?

Comment (CherryPerry) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/maybe/TestMaybeObserver.kt]: Agree.

Comment (arkivanov) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/maybe/TestMaybeObserver.kt]: I have applied the requested changes

**Comment by arkivanov** [comment]
@CherryPerry Had to fix the error text for `TestMaybeObserver`
